pkg/workerapi/workerserver/server.go

```diff
@@ -0,0 +1,64 @@
+// Package workerserver provides a way for a worker to set up servers for both
+// gRPC and HTTP communication.
+package workerserver
+
+import (
+	"net"
+
+	"github.com/iver-wharf/wharf-cmd/pkg/resultstore"
+	"github.com/iver-wharf/wharf-core/pkg/logger"
+	"github.com/soheilhy/cmux"
+)
+
+var log = logger.NewScoped("WORKER-SERVER")
+
+// Server contains both a gRPC server and an HTTP server.
+//
+// It provides an easy way to start both of these servers simultaneously,
+// listening and responding to requests to either.
+type Server struct {
+	rest *restServer
+	grpc *grpcWorkerServer
+
+	listener net.Listener
+}
+
+// Serve starts the gRPC and HTTP servers.
+func (s *Server) Serve(bindAddress string, store resultstore.Store, artifactReader ArtifactReader) error {
+	s.rest = newRestServer(artifactReader)
+	s.grpc = newGRPCServer(store)
```

Comment:
The `resultstore.Store` and `ArtifactReader` parameters are better assigned via a constructor/`New` function, as they modify the object's state.